content/docs/installation/best-practice.md

```diff
@@ -1,19 +1,102 @@
+and the [timings and thresholds can be configured using Helm values](https://github.com/cert-manager/cert-manager/blob/eafe0d0aae4b7a9411825424f6b43fb623e1ba65/deploy/charts/cert-manager/README.template.md?plain=1#L181-L185).
+
+### controller
+
+> ℹ️ The cert-manager controller liveness probe was introduced in cert-manager `v1.12.0`.
+
+The cert-manager controller has a liveness probe, but it is **disabled by default**.
+You can enable it using the Helm chart value `livenessProbe.enabled=true`,
+but first read the background information below.
+
+The liveness probe for the cert-manager controller is an HTTP probe which connects
+to the `/livez` endpoint of a healthz server which listens on port 9443 and runs in its own thread.
+The `/livez` endpoint currently reports the combined status of the following sub-systems
+and each sub-system has its own `/livez` endpoint. These are:
+
+* `/livez/leaderElection`: Returns an error if the leader election record has not been renewed
+  or if the leader election thread has exited without also crashing the parent process.
+
+> ℹ️ In future more sub-systems could be checked by the `/livez`  endpoint,
+> similar to how Kubernetes [ensure logging is not blocked](https://github.com/kubernetes/kubernetes/pull/64946)
+> and have [health checks for each controller](https://github.com/kubernetes/kubernetes/pull/104667).
+>
+> 📖 Read about [how to access individual health checks and verbose status information](https://kubernetes.io/docs/reference/using-api/health-checks/) (cert-manager uses the same healthz endpoint multiplexer as Kubernetes).
+
+### cainjector
+
+The cainjector Pod does not have a liveness probe or a `/livez` healthz endpoint,
+but there is justification for it in the GitHub issue:
+[cainjector in a zombie state after attempting to shut down](https://github.com/cert-manager/cert-manager/issues/5889).
+Please add your remarks to that issue if you have also experienced this specific problem,
+and add your remarks to [Helm: Allow configuration of readiness, liveness and startup probes for all created Pods](https://github.com/cert-manager/cert-manager/issues/5626) if you have a general request for a liveness probe in cainjector.
+
+### Background Information
+
+The cert-manager `controller` process and the `cainjector` process,
+both use the Kubernetes [leader election library](https://pkg.go.dev/k8s.io/client-go/tools/leaderelection),
+to ensure that only one replica of each process can be active at any one time.
+The Kubernetes control-plane components also use this library.
+
+The leader election code runs in a loop in a separate thread (go routine).
+If it initially wins the leader election race and if it later fails to renew its leader election lease, it exits.
```

Comment:
This is interesting, I actually didn't realize that the behaviour was like that. I imagined that if a replica loses leader election (because another replica managed to become a leader) it would just become a standby and keep attempting to become a leader. I think that's how it works in raft.
However I do see that the way we do it is also how c/r does it https://github.com/kubernetes-sigs/controller-runtime/blob/cd65cb25d314f40a329a688f4714fe3282589e97/pkg/manager/internal.go#L646 so perhaps it makes sense to exit with an error because a leader election cannot be lost in a normal flow (there's no way how leadership could shift unless there is a fatal errror). Something to look into for myself 👀